tests/unittest/test_services.py

```diff
@@ -18,6 +28,865 @@ def test_conf_expansion_cli(self):
             parsed_property.props["my-other-conf"], "/this/does/$NOT/change"
         )
 
+    def test_kube_interface(self):
+
+        # mock logic
+        test_id = str(uuid.uuid4())
+        username1 = str(uuid.uuid4())
+        context1 = str(uuid.uuid4())
+        token1 = str(uuid.uuid4())
+        username2 = str(uuid.uuid4())
+        context2 = str(uuid.uuid4())
+        token2 = str(uuid.uuid4())
+        username3 = str(uuid.uuid4())
+        context3 = str(uuid.uuid4())
+        token3 = str(uuid.uuid4())
+        test_kubectl_cmd = str(uuid.uuid4())
+
+        kubeconfig_yaml = {
+            "apiVersion": "v1",
```

Comment:
it would be best to put the yaml as a YAML file, and then read it. 

We could place the YAML file in tests/resource/ or tests/unittest/resource, either is fine by me. 

See these links for some inspiration:

https://github.com/CGnal/cgnal-core/blob/main/tests/__init__.py#L9
https://github.com/CGnal/cgnal-core/blob/main/tests/test_config.py#L55
https://github.com/CGnal/cgnal-core/tree/main/tests/resources/data

Reply:
Let's leave this out for another refactoring or improvemnt. 

This is a minor point

Reply:
Thanks Enrico. Addressed rest of the comments.